app/src/main/java/com/github/se/studentconnect/ui/profile/ProfileScreenViewModel.kt

```
@@ -121,7 +121,20 @@ class ProfileScreenViewModel(
 
     // Combine and remove duplicates
     val allEventIds = combineEventIds(joinedEventIds, createdEventIds)
-    _eventsCount.value = allEventIds.size
+
+    // Verify that events actually exist before counting them
+    val existingEventsCount =
+        allEventIds.count { eventId ->
+          try {
+            eventRepository.getEvent(eventId)
+            true
+          } catch (e: Exception) {
```

Too broad IMO. Consider catching more specific exceptions (`NoSuchElementException`) to avoid masking unexpected errors.